svf-llvm/lib/SVFIRBuilder.cpp

```diff
@@ -948,12 +948,22 @@ void SVFIRBuilder::visitSwitchInst(SwitchInst &inst)
     BranchStmt::SuccAndCondPairVec successors;
     for (u32_t i = 0; i < inst.getNumSuccessors(); ++i)
     {
-        const Instruction* succInst = &inst.getSuccessor(i)->front();
-        const ConstantInt* condVal = inst.findCaseDest(inst.getSuccessor(i));
+        BasicBlock *succBB = inst.getSuccessor(i);
         /// default case is set to -1;
         s64_t val = -1;
-        if (condVal && condVal->getBitWidth() <= 64)
-            val = condVal->getSExtValue();
+        /// Equivalent to: succBB != inst.getDefaultDest()
+        if (i != 0)
+        {
+            // According to llvm::SwitchInst class comments:
+            // Operand[0]    = Value to switch on
+            // Operand[1]    = Default basic block destination
+            // Operand[2n  ] = Value to match
+            // Operand[2n+1] = BasicBlock to go to on match
+            ConstantInt *condVal = SVFUtil::dyn_cast<ConstantInt>(inst.getOperand(i * 2));
```

Comment:
what does this mean here, why i * 2? Can we use `findCaseDest` + something else?

Reply:
In my mind, 'condVal' is target `case value`. And in LLVM IR:

Operand 0 is `condition of switch`, 
Operand 1 is `default` BasicBlock Value , and 

Operand 2 is target Value of `the first` Case
Operand 3 is target succussor BasicBlock Value of `the first` Case's 
...
Operand 2n      is target Value of `the n-th` Case
Operand 2n+1  is target succussor BasicBlock Value of `the n-th` Case's

<hr/>

Before we use findCaseDest, source code of findCaseDest is below: 
```c
  /// Finds the unique case value for a given successor. Returns null if the
  /// successor is not found, not unique, or is the default case.
  ConstantInt *findCaseDest(BasicBlock *BB) {
    if (BB == getDefaultDest())
      return nullptr;

    ConstantInt *CI = nullptr;
    for (auto Case : cases()) {
      if (Case.getCaseSuccessor() != BB)
        continue;

      if (CI)
        return nullptr; // Multiple cases lead to BB.

      CI = Case.getCaseValue();
    }

    return CI;
  }
```
we will get -1 findCaseDest of default case, and also get -1 when encounter continuous case without break stmt, the LLVM IR related the given code is 
```c
// ... 
  switch i32 %0, label %sw.default [
    i32 0, label %sw.bb                                ; get -1 due to default branch
    i32 1, label %sw.bb1                               ; get -1 without break
    i32 2, label %sw.bb1                              ; get -1  without break
    i32 3, label %sw.bb1                             
    i32 4, label %sw.bb3                             ; get -1 without break
    i32 6, label %sw.bb3                             ; get -1 without break
    i32 7, label %sw.bb3
  ], !dbg !18
// ... 
```

if we use `findCaseDest` + something else, let me thought a few moments. I can't give the answer at once...